### PR TITLE
Fix subshell syntax error

### LINKED ISSUE #136: Fix syntax error
## Example

(ls ))

### PR DESCRIPTION
**Pull request by alex216**
fix #136 

## Release Notes

- **New Features**
  - Enhanced command parsing logic for improved error handling and control flow.
  - Introduced new functions to validate command-following tokens across various parsing components.

- **Bug Fixes**
  - Improved handling of syntax errors in subshell commands, ensuring accurate error reporting for various edge cases.

- **Tests**
  - Added comprehensive test cases for subshell functionality, including both successful executions and expected syntax errors.

- **Documentation**
  - Updated author information and timestamps in several files for better tracking of contributions.

### CODE REVIEW

**Review by rask24 — APPROVED** [review]
@alex216 LGTM 🎲

File: src/parser/parse_list.c [hunk 1]
```diff
@@ -29,15 +37,17 @@
...
+			return (destroy_ast(node));
 		node_type = (t_ast_node_type)get_token_type(*cur_token);
-		if (node_type != AST_AND && node_type != AST_OR)
+		if (!(node_type == AST_AND || node_type == AST_OR))
 			break ;
 		consume_token(cur_token);
 		tmp = parse_pipeline(cur_token);
 		if (tmp == NULL)
-			return (NULL);
+			return (destroy_ast(node));
```
[the remark this answers is not in the document]
Comment (alex216) [reply src/parser/parse_list.c:50]: This code is actual won't be called, but it's there for code readability, just watch and learn, bro.
